urllib3/connectionpool.py

```diff
@@ -666,6 +666,10 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                                 release_conn=release_conn, body_pos=body_pos,
                                 **response_kw)
 
+        def drain_and_release_conn(response):
+            response.read()
```

Comment:
Ah, I see. Yeah, we probably need most of those. `CertificateError` we don't need, but the rest seem like good things to catch.